hs_core/views/utils.py

```diff
@@ -57,18 +58,22 @@ def authorize(request, res_id, edit=False, view=False, full=False, superuser=Fal
     except ObjectDoesNotExist:
         raise NotFound(detail="No resource was found for resource id:%s" % res_id)
 
-    if user.is_authenticated():
+    if discoverable and (edit is False and view is False and full is False and superuser is False):
+        authorized = res.raccess.discoverable
+    elif user.is_authenticated():
         has_edit = user.uaccess.can_change_resource(res)
         has_view = user.uaccess.can_view_resource(res)
```

Comment:
@selimnairb Yes, that is what I think also. I think it is perhaps better to add an additional check after authorize() call when needing to determine whether content files are viewable or not by just checking whether it is a resource with discoverable set to true but public set to false.